### PR TITLE
Replace hipblas by rocblas

### PR DESCRIPTION
**Pull request by aurianer**
Necessary as first step for https://github.com/eth-cscs/DLA-Future/issues/577 to avoid redefinition of the macros.

Will wait for #67 to be merged.

### CODE REVIEW

**Comment by aurianer** [comment]
bors try

**Comment by aurianer** [comment]
bors try

File: libs/pika/async_cuda/tests/unit/then_with_stream.cu
Comment (msimberg) [inline libs/pika/async_cuda/tests/unit/then_with_stream.cu:128]: With this change the expected call counts will need to be updated for HIP since the `cusolverDnHandle_t` overloads will never be called.

Comment (aurianer) [reply libs/pika/async_cuda/tests/unit/then_with_stream.cu:128]: Thanks, yes I went too fast

File: libs/pika/async_cuda/tests/unit/then_with_stream.cu
Comment (msimberg) [inline libs/pika/async_cuda/tests/unit/then_with_stream.cu:466]: As already mentioned, let's wait for #67 to go in before this. It's not sufficient to just remove `cusolver_*_calls` checks. The counts from those will be distributed among the `host_*_calls`, `stream_*_calls`, `cublas_*_calls` variables.

**Review by msimberg — APPROVED** [review]
The `then_with_stream` test still needs updating, but this looks good to me otherwise.

File: libs/pika/async_cuda/tests/unit/then_with_stream.cu
Comment (aurianer) [reply libs/pika/async_cuda/tests/unit/then_with_stream.cu]: I updated the counts, should be good now

**Comment by aurianer** [comment]
bors try

File: libs/pika/async_cuda/src/cublas_exception.cpp
Comment (msimberg) [inline libs/pika/async_cuda/src/cublas_exception.cpp:26]: Should we move the "common" error codes separately into the `PIKA_HAVE_HIP` and `PIKA_HAVE_CUDA` sections so that the returned strings are correct? I.e. so that we can have `rocblas_status_invalid_value` printed with HIP enabled instead of `CUBLAS_STATUS_INVALID_VALUE` etc.?

Comment (msimberg) [reply libs/pika/async_cuda/src/cublas_exception.cpp:26]: Note, this comment does not invalidate my previous approval. Can be changed separately.

Comment (aurianer) [reply libs/pika/async_cuda/src/cublas_exception.cpp:26]: yes definitely!

Comment (aurianer) [reply libs/pika/async_cuda/src/cublas_exception.cpp]: I updated it in this PR

**Review by msimberg — APPROVED** [review]
Thanks @aurianer.

This is ready to be merged. However, as already discussed separately I'd like to time merging this close to merging the corresponding changes to DLA-Future. Let's wait for that to be close to ready before actually pressing the merge button.

**Comment by msimberg** [comment]
I think with 0.8.0 just released we don't actually need to wait any longer to merge this. We have all the thus-far known required fixes for CP2K in 0.8.0 so CP2K can stop using pika main and stick to 0.8.0.

@aurianer do you have further changes you'd like to have in here before merging?

**Comment by msimberg** [comment]
bors merge

**Comment by msimberg** [comment]
bors merge

**Pull request merged**
